### PR TITLE
Change server actions cache default to no-store

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Currently there is a bug in Server Actions when you `fetch` as it uses the same defaults (caching when not specified) as rendering, this causes some issues as you want to read your writes in Server Actions.

This change adds the `no-store` default for Server Actions, you can still override it by specifying `cache: 'force-cache'` for example, but it defaults to `cache: 'no-store'`.

Fixes NEXT-1926

### CODE REVIEW

**Comment by timneutkens** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #60170** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60170?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/60170?utm_source=stack-comment).

File: packages/next/src/server/app-render/action-handler.ts [hunk 1]
```diff
@@ -299,6 +299,9 @@ export async function handleAction({
     )
   }
 
+  // When running actions the default is no-store, you can still `cache: 'force-cache'`
```
Comment (feedthejim) [inline packages/next/src/server/app-render/action-handler.ts:302]: can you add a test for the force-cache scenario>

**Review by feedthejim — CHANGES_REQUESTED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 53f8f973a90852e22358fabf2a01b53debb4c045

`pnpm test test/integration/prerender-export/test/index.test.js` 

<details>
<summary>Expand output</summary>



  ● Test suite failed to run

    thrown: "Exceeded timeout of 60000 ms for a hook.
    Add a timeout value to this test to increase the timeout, if this is a long-running test. See https://jestjs.io/docs/api#testname-fn-timeout."

      217 |       })
      218 |
    > 219 |       afterAll(async () => {
          |       ^
      220 |         if (app) {
      221 |           await stopApp(app)
      222 |         }

      at afterAll (integration/prerender-export/test/index.test.js:219:7)
      at describe (integration/prerender-export/test/index.test.js:209:5)
      at integration/prerender-export/test/index.test.js:208:52
      at Object.describe (integration/prerender-export/test/index.test.js:207:1)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| buildDuration | 12.8s | 12.9s | N/A |
| buildDurationCached | 7.1s | 6.1s | N/A |
| nodeModulesSize | 202 MB | 202 MB | ⚠️ +3.42 kB |
| nextStartRea..uration (ms) | 437ms | 426ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 433-HASH.js gzip | 28.4 kB | 28.5 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 239 B | 242 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.5 kB | 98.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.28 kB | 4.28 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| index.html gzip | 528 B | 526 B | N/A |
| link.html gzip | 541 B | 538 B | N/A |
| withRouter.html gzip | 524 B | 521 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.8 kB | N/A |
| page.js gzip | 147 kB | 147 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 623 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.4 kB | 37.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.07 kB | 2.07 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-03-Change_server_actions_cache_default_to_no-store | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 168 kB | 168 kB | N/A |
| app-page-exp..prod.js gzip | 94.2 kB | 94.2 kB | N/A |
| app-page-tur..prod.js gzip | 94.9 kB | 94.9 kB | N/A |
| app-page-tur..prod.js gzip | 89.4 kB | 89.4 kB | N/A |
| app-page.run...dev.js gzip | 138 kB | 138 kB | N/A |
| app-page.run..prod.js gzip | 88.7 kB | 88.8 kB | N/A |
| app-route-ex...dev.js gzip | 24.1 kB | 24.1 kB | ✓ |
| app-route-ex..prod.js gzip | 16.7 kB | 16.7 kB | ✓ |
| app-route-tu..prod.js gzip | 16.7 kB | 16.7 kB | ✓ |
| app-route-tu..prod.js gzip | 16.3 kB | 16.3 kB | ✓ |
| app-route.ru...dev.js gzip | 23.5 kB | 23.5 kB | ✓ |
| app-route.ru..prod.js gzip | 16.3 kB | 16.3 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.38 kB | 9.38 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.65 kB | 9.65 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.4 kB | 49.4 kB | ✓ |
| Overall change | 258 kB | 258 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 53f8f973a90852e22358fabf2a01b53debb4c045

File: packages/next/src/server/app-render/action-handler.ts (hunk 1 above)
Comment (timneutkens) [reply packages/next/src/server/app-render/action-handler.ts:302]: Done!

**Review by feedthejim — APPROVED** [review]

File: test/e2e/app-dir/actions/app-action.test.ts [hunk 2]
```diff
@@ -1114,5 +1114,77 @@ createNextDescribe(
...
+        const getNumber = async () =>
+          JSON.parse(await browser.elementByCss('#fetched-data').text())
+
+        const firstNumber = await getNumber()
+
+        await browser.waitForElementByCss('#trigger-fetch').click()
+
+        await check(async () => {
+          const newNumber = await getNumber()
+          // Expect that the number changes on each click
```
Comment (ztanner) [inline test/e2e/app-dir/actions/app-action.test.ts:1158]:
```suggestion
          // Expect that the number is the same on each click
```

File: test/e2e/app-dir/actions/app-action.test.ts [hunk 3]
```diff
@@ -1114,5 +1114,77 @@ createNextDescribe(
...
```
Comment (ztanner) [inline test/e2e/app-dir/actions/app-action.test.ts:1182]:
```suggestion
          // Expect that the number is the same on each click
```

**Pull request merged**